### PR TITLE
PX-907 / Simplify Phone Input

### PR DESCRIPTION
**Pull request by jmhar3**
## What is the desired outcome?
Modify the PhoneInput to allow for simpler options.

## Why do we want this?
Patrick wants a phone input with the prefix + suffix removed.

## How?
Make prefix + suffix optional.

## Testing?
Tests have been added but you can run storybook locally to try it for yourself

## Screenshots
N/A (see below)

## Anything else? Questions? Concerns?
I'm getting an error when attempting to run storybook so I haven't been able to preview locally.

### CODE REVIEW

**Review by lukerollans — DISMISSED** [review]
@jmhar3 I think there's just a small handful of things to take care of here

## Checklist
1. Absorb `showPrefix` and `showSuffix` in to a `format: "international" | "local"` prop
2. When in `"international"` mode, report back the number inclusive of international code
3. When in `"local"` mode, report back whatever the user enters while still restricting input to numbers only.
4. Fix Chakra UI storybook styling. Either upgrade to fix the Chakra UI Storybook addon, or add ChakraProvider to each story inline

## Implementation
I think I'm starting to remember why we opted not to use `libphonenumber-js` - we only allow AU or US numbers, so a global validation library addition to the bundle wasn't justifiable.

Given we already have `maxLength` set, hopefully this won't be too much effort

File: src/components/forms/PhoneInput.tsx [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import * as React from "react"
+import parsePhoneNumber from "libphonenumber-js"
```
Comment (sumac13) [inline src/components/forms/PhoneInput.tsx]: I think the linter here wants you to use 
```suggestion
import { parsePhoneNumber } from "libphonenumber-js"
```

File: src/components/forms/PhoneInput.tsx [hunk 2]
```diff
@@ -97,9 +109,20 @@ export const PhoneInput = (props: PhoneInputProps) => {
...
+        }${value}`
+
         if (onChange) {
-          onChange(`${currentCountry.callingCode}${value}`)
+          onChange(phoneNumber)
+        }
+
+        if (onValidatePhoneNumber) {
+          onValidatePhoneNumber(
+            parsePhoneNumber(phoneNumber, currentCountry.countryCode).isValid()
```
Comment (sumac13) [inline src/components/forms/PhoneInput.tsx]: couple of lint issues that are worth looking at too

File: stories/components/forms/PhoneInput.stories.tsx [hunk 3]
```diff
@@ -36,3 +38,9 @@ Placeholder.args = {
   initialCountryCode: CountryCode.AU,
   showPlaceholderExampleNumber: true,
 }
+
+HideCountryCodeAffixes.args = {
+  format: Format.LOCAL,
+  showPrefix: false,
```
Comment (sumac13) [inline stories/components/forms/PhoneInput.stories.tsx]: should these `show...` flags be removed if we are moving to the Format param instead?

File: src/components/forms/PhoneInput.tsx [hunk 4]
```diff
@@ -97,9 +109,20 @@ export const PhoneInput = (props: PhoneInputProps) => {
       if (value === "" || /^[0-9\b]+$/.test(value)) {
         // Persist the internally controlled value
         setNumber(value)
+
         // Format the number as E164 and send it along
+        const phoneNumber = `${
```
Comment (sumac13) [inline src/components/forms/PhoneInput.tsx]: I think there is a method in the `libphonenumber-js` lib that will output this for you without needing to do string templating like this and ensures it handles all edge cases.

Something like 
```
const phoneNumber = isInternational ? parsePhoneNumber(value, currentCountry.countryCode).number : value
```

**Review by sumac13 — CHANGES_REQUESTED** [review]

**Review by sumac13 — APPROVED** [review]

**Pull request merged**